client/src/components/ContentNavigator/utils.ts

```diff
@@ -106,3 +108,64 @@ export const isItemInRecycleBin = (item: ContentItem): boolean =>
   !!item && item.flags?.isInRecycleBin;
 
 export const isContentItem = (item): item is ContentItem => isValidItem(item);
+
+/**
+ * This attempts to create a new variabe in our SAS program using the following rules:
+ * - First, we attempt to use only the alphanumeric characters from contentItem.name as
+ *   as our variable name (without the file extension).
+ * - If that's empty, we instead use `<extension>file`
+ * - Next, we add a numeric suffix to the variable name, and check documentContent to make
+ *   sure what we're creating isn't already in use.
+ * - If the current variable is in use, we continue to increment our numeric suffix until
+ *   we find one that doesn't exist in our document.
+ *
+ * @param contentItemName ContentItem.name used for creating variable name.
+ * @param documentContent The contents of the editor we're dropping into.
+ *
+ * @returns string our new variable name.
+ */
+const extractVariableName = (
+  contentItemName: string,
+  documentContent: string,
+): string => {
+  const filePieces = parse(contentItemName);
+  const partialFileName = (
+    filePieces.name.replace(/[^a-zA-Z0-9]/g, "") ||
+    `${filePieces.ext.replace(".", "")}file`
+  ).toLocaleLowerCase();
+
+  let idx = 0;
+  const lowercaseDocumentContent = documentContent.toLocaleLowerCase();
+  while (
+    // The trailing space is intentional here so we don't confuse things like
+    // `csvfile11` for `csvfile1`
+    lowercaseDocumentContent.indexOf(`${partialFileName}${++idx} `) !== -1
+  ) {
+    /* empty */
+  }
+
+  return `${partialFileName}${idx}`;
+};
+
+// A document uses uppercase letters _if_ there is more than one
+// word (where word means gte 3 characters) that is all uppercase
+const documentUsesUppercase = (documentContent: string) =>
```

Comment:
Well, if you try with the `client/testFixture/SampleCode.sas`, it will insert uppercase. Because there're datalines with uppercase but not code.
Looks challenging.
How about only do uppercase if not found lowercase word?

Reply:
Updated in https://github.com/sassoftware/vscode-sas-extension/pull/510/commits/1e445783cf30d8ac6246174bf0b17a6bc8f5b209